common/playlist.c

```diff
@@ -178,6 +179,27 @@ void playlist_shuffle(struct playlist *pl)
     talloc_free(arr);
     pl->current = save_current;
     pl->current_was_replaced = save_replaced;
+    pl->shuffled = true;
+}
+
+void playlist_unshuffle(struct playlist *pl) {
+    struct playlist_entry *save_current = pl->current;
+    bool save_replaced = pl->current_was_replaced;
+    int count = playlist_count(pl);
+    struct playlist_entry **arr = talloc_array(NULL, struct playlist_entry *,
+                                               count);
+    for (int n = 0; n < count; n++) {
+        arr[pl->first->pos] = pl->first;
```

Comment:
If I ever get around to it, I was eventually planning to make it so that you could unshuffle even if you added elements whilst the playlist was shuffled, although that would depend on what we think of adding videos to a shuffled playlist, do they reshuffle the playlist, do they go in the end, do they get added in a random position. (As in, what should the behaviour be?)